File: easy/2/2.pl
% compile: ['2.pl']
% start with: start.

start:-
    format("What would you like to find? type Type f for force, m for mass, or a for Acceleration: ~n", []),
    ask_function(Func),
    find(Func).


ask_function(Func):-
    read(Func),
    nl.

find(f):-
    format("What is the acceleration? ~n", []),
    read(Acc),
    format("What is the mass? ~n", []),
    read(Mass),
    F is Acc*Mass,
    format("The force is: ~w ~n", [F]).

find(f):-
    format("What is the acceleration? ~n", []),
    read(Acc),
    format("What is the mass? ~n", []),
    read(Mass),
    F is Acc*Mass,
    format("The force is: ~w ~n", [F]).

find(m):-
    format("What is the acceleration? ~n", []),
    read(Acc),
    format("What is the force? ~n", []),
    read(F),
    M is F/Acc,
    format("The mass is: ~w ~n", [M]).

find(a):-
    format("What is the mass? ~n", []),
    read(M),
    format("What is the force? ~n", []),
    read(F),
    A is F/M,
    format("The acceleration is: ~w ~n", [A]).
    

File: easy/2/2.erl
% c('2.erl').
% '2'.start().

-module('2').

-compile(export_all).


start()->
    io:format("What would you like to find? type Type F for force, M for mass, or A for Acceleration: ~n"),
    Choice = hd(string:tokens(io:get_line("> "),"\r\n")),
    case Choice of
        "A" ->
            F = hd(string:tokens(io:get_line("What is the force? "),"\r\n")),
            M = hd(string:tokens(io:get_line("What is the mass? "),"\r\n")),
            acc(list_to_integer(F),list_to_integer(M));
        "F" ->
            A = hd(string:tokens(io:get_line("What is the acceleration? "),"\r\n")),
            M = hd(string:tokens(io:get_line("What is the mass? "),"\r\n")),
            force(M,A);
        "M" ->
            F = hd(string:tokens(io:get_line("What is the force? "),"\r\n")),
            A = hd(string:tokens(io:get_line("What is the acceleration? "),"\r\n")),
            mass(list_to_integer(F),list_to_integer(A));
        true -> io:format("Invalid input ~n")
                end.


mass(F,A)->
    M = F/A,
    io:format("The mass is: ~w ~n", [M]).

acc(F,M)->
    A = F/M,
    io:format("The acceleration is: ~w ~n", [A]).

force(M,A)->
    F = M*A,
    io:format("The force is: ~w ~n", [F]).
